### PR TITLE
TIP-1051: Add GetValuesAndPropertiesFromProductIdentifiers

### PR DESCRIPTION
**Pull request by pierallard**
| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

### CODE REVIEW

File: src/Akeneo/Pim/Enrichment/Bundle/Product/Query/Sql/GetValuesAndPropertiesFromProductIdentifiers.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+use Doctrine\DBAL\Connection;
+
+/**
+ * @author Pierre Allard <pierre.allard@akeneo.com>
+ */
+class GetValuesAndPropertiesFromProductIdentifiers
+{
+    /** @var Connection */
+    var $connection;
```
Comment (anaelChardan) [inline src/Akeneo/Pim/Enrichment/Bundle/Product/Query/Sql/GetValuesAndPropertiesFromProductIdentifiers.php]: I think you do to much javascript :D

File: src/Akeneo/Pim/Enrichment/Bundle/Product/Query/Sql/GetValuesAndPropertiesFromProductIdentifiers.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+        $rows = $this->connection->fetchAll(
+            $query,
+            [$productIdentifiers],
+            [Connection::PARAM_STR_ARRAY]
+        );
+
+        $results[] = [];
+        foreach ($rows as $row) {
+            $results[$row['product_identifier']] = $row;
```
Comment (anaelChardan) [inline src/Akeneo/Pim/Enrichment/Bundle/Product/Query/Sql/GetValuesAndPropertiesFromProductIdentifiers.php]: You have to alias then ;)

File: src/Akeneo/Pim/Enrichment/Bundle/Product/Query/Sql/GetValuesAndPropertiesFromProductIdentifiers.php [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        );
+
+        $results = [];
+        foreach ($rows as $row) {
+            $results[$row['identifier']] = [
+                'identifier' => $row['identifier'],
+                'is_enabled' => $row['is_enabled'] === '1',
+                'product_model_code' => $row['product_model_code'],
+                'created' => \DateTime::createFromFormat('Y-m-d H:i:s', $row['created']),
+                'updated' => \DateTime::createFromFormat('Y-m-d H:i:s', $row['updated']),
```
Comment (ahocquard) [inline src/Akeneo/Pim/Enrichment/Bundle/Product/Query/Sql/GetValuesAndPropertiesFromProductIdentifiers.php]: create it by using the dbal https://github.com/ahocquard/pim-community-dev/blob/ade08816847d36eee74dbb058673d3574f136e15/src/Pim/Bundle/ResearchBundle/Infrastructure/Persistence/Database/DatabaseFamilyRepository.php#L100

File: src/Akeneo/Pim/Enrichment/Bundle/Product/Query/Sql/GetValuesAndPropertiesFromProductIdentifiers.php [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+                'family_code' => $row['family_code'],
+                'group_codes' => array_filter(json_decode($row['group_codes']), function ($groupCode) {
```
Comment (ahocquard) [inline src/Akeneo/Pim/Enrichment/Bundle/Product/Query/Sql/GetValuesAndPropertiesFromProductIdentifiers.php]: no need to have a function, array_filter filters the null values by default

File: src/Akeneo/Pim/Enrichment/Bundle/Product/Query/Sql/GetValuesAndPropertiesFromProductIdentifiers.php [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+                    return null !== $groupCode;
+                }),
+                'raw_values' => json_decode($row['raw_values'], true)
```
Comment (ahocquard) [inline src/Akeneo/Pim/Enrichment/Bundle/Product/Query/Sql/GetValuesAndPropertiesFromProductIdentifiers.php]: you should return a value collection

File: src/Akeneo/Pim/Enrichment/Bundle/Product/Query/Sql/GetValuesAndPropertiesFromProductIdentifiers.php [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+/**
+ * SQL Query to get the properties and the values from a set of product identifiers:
+ * identifier, is_enabled, product_model_code, created, updated, family_code, group_codes and raw_data.
+ *
+ * @author    Pierre Allard <pierre.allard@akeneo.com>
+ * @copyright 2019 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
+ */
+class GetValuesAndPropertiesFromProductIdentifiers
```
Comment (ahocquard) [inline src/Akeneo/Pim/Enrichment/Bundle/Product/Query/Sql/GetValuesAndPropertiesFromProductIdentifiers.php]: final

File: src/Akeneo/Pim/Enrichment/Bundle/Product/Query/Sql/GetValuesAndPropertiesFromProductIdentifiers.php (hunk 4 above)
Comment (anaelChardan) [reply src/Akeneo/Pim/Enrichment/Bundle/Product/Query/Sql/GetValuesAndPropertiesFromProductIdentifiers.php]: Not sure if it has to be done here. Imho no as we use return light things

File: src/Akeneo/Pim/Enrichment/Bundle/Product/Query/Sql/GetValuesAndPropertiesFromProductIdentifiers.php (hunk 5 above)
Comment (anaelChardan) [reply src/Akeneo/Pim/Enrichment/Bundle/Product/Query/Sql/GetValuesAndPropertiesFromProductIdentifiers.php]: Not sure if it has to be done here. Imho no as we use return light things

Comment (ahocquard) [reply src/Akeneo/Pim/Enrichment/Bundle/Product/Query/Sql/GetValuesAndPropertiesFromProductIdentifiers.php]: Nanou convinced me (after a lot of discussion). Let's keep it like that for now as it's infra stuff.

File: src/Akeneo/Pim/Enrichment/Bundle/Product/Query/Sql/GetValuesAndPropertiesFromProductIdentifiers.php [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+{
+    /** @var Connection */
+    private $connection;
+
+    public function __construct(Connection $connection)
+    {
+        $this->connection = $connection;
+    }
+
+    public function fetchByProductIdentifiers(array $productIdentifiers)
```
Comment (anaelChardan) [inline src/Akeneo/Pim/Enrichment/Bundle/Product/Query/Sql/GetValuesAndPropertiesFromProductIdentifiers.php]: misses return type

Comment (pierallard) [reply src/Akeneo/Pim/Enrichment/Bundle/Product/Query/Sql/GetValuesAndPropertiesFromProductIdentifiers.php]:
```suggestion
    public function fetchByProductIdentifiers(array $productIdentifiers): array
```

**Pull request merged**